fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java

```diff
@@ -1029,4 +1047,95 @@ private static BigInteger bitShiftRightLogicalForInt128(BigInteger l, int shiftB
         BigInteger opened = l.subtract(INT_128_OPENER);
         return opened.shiftRight(shiftBy).and(INT_128_MASK1_ARR1[shiftBy]);
     }
+
+    // =================================== meta functions ==================================== //
+
+    /**
+     * Return verbose metadata of a materialized-view
+     */
+    @ConstantFunction(name = "inspect_mv_meta", argTypes = {VARCHAR}, returnType = VARCHAR, isMetaFunction = true)
+    public static ConstantOperator inspect_mv_meta(ConstantOperator mvName) {
+        TableName tableName = TableName.fromString(mvName.getVarchar());
+        Table table = GlobalStateMgr.getCurrentState()
+                .mayGetDb(tableName.getDb())
+                .flatMap(db -> db.mayGetTable(tableName.getTbl()))
+                .orElseThrow(
+                        () -> ErrorReport.buildSemanticException(ErrorCode.ERR_BAD_TABLE_ERROR, mvName.getVarchar()));
+        if (!table.isMaterializedView()) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_INVALID_PARAMETER,
+                    mvName.getVarchar() + " is not materialized view");
+        }
+        ConnectContext connectContext = ConnectContext.get();
+        PrivilegeChecker.checkAnyActionOnTable(
```

Comment:
there are many duplicate code in the following function, could you construct a more common framework to make adding new function more easy?

Reply:
fixed.